src/ai.ts

```diff
@@ -0,0 +1,126 @@
+import * as core from "@actions/core";
+import { File as DiffFile } from "parse-diff";
+import { ReviewComment } from "./types";
+import OpenAI from "openai";
+import Anthropic from "@anthropic-ai/sdk";
+
+const INFERENCE_API_KEY = core.getInput("INFERENCE_API_KEY");
+const MODEL = core.getInput("MODEL");
+// TODO: Conditional import for openai and anthropic
+const openai = new OpenAI({ apiKey: INFERENCE_API_KEY });
+const customInstructions = core.getInput("custom_instructions");
+
+export function createReviewPrompt(
+  prContext: { title: string; description: string; changedFiles: string[] },
+  repoContext: { readmeContent: string; packageJsonContent: string },
+  file: DiffFile,
+  content: string
+): string {
+  const diffReviews = file.chunks
+    .map(
+      (chunk, index) => `Diff ${index + 1}:
+${chunk.content}
+${chunk.changes
+  .map((change) => {
+    // @ts-expect-error - ln and ln2 exist where needed
+    const lineNumber = change.ln || change.ln2;
+    return `${lineNumber} ${change.content}`;
+  })
+  .join("\n")}
+`
+    )
+    .join("\n");
+
+  console.debug(diffReviews);
+  return `Review the following code diffs in the file "${file.to}":
+
+PR Title: ${prContext.title}
+PR Description: ${prContext.description}
+
+Git diffs to review:
+---
+${diffReviews}
+---
+
+Full file content:
+---
+${content}
+---
+
+Provide a code review in JSON format:
+{
+  "comments": [
+    {
+      "diffIndex": <diff_index>,
+      "lineNumber": <line_number>,
+      "comment": "<review comment>"
+    }
+  ]
+}
+
+Specifically your instructions are:
+- Focus on code quality, best practices, and potential issues.
+- Only provide comments on code that has changed.
```

Comment:
The new instruction to only provide comments on changed code is redundant. This behavior is already implied by the nature of a code review for a pull request. Consider removing this instruction to avoid confusion and maintain a concise set of guidelines.